sirepo/template/srw.py

```diff
@@ -2270,6 +2251,42 @@ def _wavefront_pickle_filename(el_id):
     return 'initial.pkl'
 
 
+def _write_rsopt_zip(data, ctx):
+
+    def _files():
+        files = []
+        for t in ('py', 'sh', 'yml',):
+            f = f'{_RSOPT_EXPORT_BASE_NAME}.{t}'
+            ctx[f'{t}FileName'] = f
+            files.append(f)
+        return files
+
+    filename = f'{_RSOPT_EXPORT_BASE_NAME}.zip'
+    with zipfile.ZipFile(
+        filename,
+        mode='w',
+        compression=zipfile.ZIP_DEFLATED,
+        allowZip64=True,
+    ) as z:
+        # the shell script depends on the other filenames being defined
+        for f in _files():
+            z.writestr(
```

Comment:
Encapsulate in a function:

```py
def _write(zip, path):
    zip.writestr(
        path,
        python_source_for_model(data, ctx.fileBase, plot_reports=False) if path.endswith('.py')
            else template_common.render_jinja(SIM_TYPE, ctx, path)
     )
```

The loop will be simpler and easier to understand.
         
        d =  python_source_for_model(data, ctx.fileBase, plot_reports=False)